client/src/components/UpcomingInterviewTable.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect, useContext } from 'react';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import { useStyles } from '../themes/theme';
+import { getStandardTime } from '../utils/timeFunctions'
+import { CustomButton } from './Buttons'
+import { useHistory } from 'react-router';
+import { cancelInterview } from '../utils/apiFunctions';
+import { fetchInterviews } from '../utils/fetchInterviews';
+import { store } from '../context/store';
+import Typography from '@material-ui/core/Typography';
+
+const UpcomingInterviewTable = ({ interviews }) => {
+  const [upcomingInterviews, setUpcomingInterviews] = useState([]);
+  const { state } = useContext(store);
+
+  useEffect(() => {
+    createUpcomingInterviewsTable(interviews);
+  }, []);
+
+  const cancelInterviewById = async (roomId) => {
+    await cancelInterview(roomId);
+    const interviews = await fetchInterviews(state.user._id);
```

Comment:
Can we just update the state here in this component to remove that interview and not make a new API call?

Reply:
Hey I'm not sure I'm following this. Would you like me to manually delete the interview from the array and then set state again?

Reply:
Yeah! The goal is to avoid any API calls that aren't fully necessary. 
```suggestion
const cancelInterviewById = async (roomId, index) => {
    const res = await cancelInterview(roomId);
    if (res.status === 200){
      const interviewsCopy = upcomingInterviews;
      interviewsCopy.splice(index, 1);
      setUpcomingInterviews(interviewsCopy);
    }
 }
  ...
  <CustomButton onClick={() => cancelInterviewById(interview.roomId, i)} ...
```

Reply:
Oh okay gotcha!